server/src/main/java/org/elasticsearch/action/admin/indices/create/TransportCreatePartitionsAction.java

```diff
@@ -201,18 +201,27 @@ private ClusterState executeCreateIndices(ClusterState currentState, CreateParti
             Map<String, AliasMetadata> templatesAliases = new HashMap<>();
             List<String> templateNames = new ArrayList<>();
 
-            List<IndexTemplateMetadata> templates = findTemplates(request, currentState);
-            applyTemplates(mapping, templatesAliases, templateNames, templates);
+
+            // We always have only 1 matching template per pattern/table.
+            // All indices in the request are related to a concrete partitioned table and
+            // they all match the same template. Thus, we can use any of them to find matching template.
+            String firstIndex = indicesToCreate.get(0);
```

Comment:
Could `indicesToCreate` be empty here?

Reply:
No, we have 
```
if (indicesToCreate.isEmpty()) {
    return currentState;
}
```
short circuit check above